ocf_data_sampler/torch_datasets/datasets/pvnet_uk.py

```diff
@@ -168,6 +166,36 @@ def get_gsp_locations(gsp_ids: list[int] | None = None) -> list[Location]:
     return locations
 
 
+def validate_dataset_channels(datasets_dict: dict, config: Configuration) -> None:
+    """Validate channels for NWP and satellite data in the dataset.
+    
+    Args:
+        datasets_dict: Dictionary containing the datasets to validate
+        config: Configuration object containing provider information
+        
+    Raises:
+        ValueError: If there's a mismatch between data channels and normalisation constants
+    """
+    if "nwp" in datasets_dict:
+        for nwp_key, da_nwp in datasets_dict["nwp"].items():
+            provider = config.input_data.nwp[nwp_key].provider
+            validate_channels(
+                data_channels=set(da_nwp.channel.values),
```

Comment:
can you remove `set` as this is done in other functions. Similar for other calls of `validate_channels`